### PR TITLE
Added json, checkstyle and code climate output formats

### LINKED ISSUE #20: Code Climate Engine
Would be useful to run `hadolint` as a Code Climate Engine:
http://blog.codeclimate.com/blog/2015/07/07/build-your-own-codeclimate-engine/

### LINKED ISSUE #73: Feature Request: Checkstyle
It would be great if you could produce checkstyle formated output. it would make this tool way better to intigrate into Jenkins e.g. and my full workflow.

### PR DESCRIPTION
**Pull request by lorenzo**
This pull request adds the ability to output hadolint's results in 4 different flavours:

* plain text
* json (with same format as shellcheck)
* codeclimate json - closes #20 
* checkstyle xml - closes #73

### CODE REVIEW

**Comment by zemanlx** [comment]
Is it on purpose that `codeclimate` is not showing issues detected by _shellcheck_?

```bash
 ~/git/hadolint   output-formats  cat >Dockerfile.test                                                                                                                                                                                
FROM node
RUN echo $THROW_ERROR
```

_ShellCheck_ JSON
```bash
 ~/git/hadolint   output-formats  /home/zeman/.local/bin/hadolint -f json Dockerfile.test | jq
[
  {
    "line": 1,
    "code": "DL3006",
    "message": "Always tag the version of an image explicitly.",
    "column": 1,
    "file": "Dockerfile.test",
    "level": "warning"
  },
  {
    "line": 2,
    "code": "SC2086",
    "message": "Double quote to prevent globbing and word splitting.",
    "column": 1,
    "file": "Dockerfile.test",
    "level": "info"
  }
]
```

Code Climate JSON
```bash
 ~/git/hadolint   output-formats  /home/zeman/.local/bin/hadolint -f codeclimate Dockerfile.test | jq
{
  "location": {
    "path": "Dockerfile.test",
    "lines": {
      "begin": 1,
      "end": 1
    }
  },
  "severity": "major",
  "check_name": "DL3006",
  "categories": [
    "Bug Risk"
  ],
  "type": "issue",
  "description": "Always tag the version of an image explicitly."
}                            
```

File: app/Main.hs [hunk 1]
```diff
@@ -44,22 +56,49 @@ instance Yaml.FromJSON ConfigFile
...
 
-printChecks :: [RuleCheck] -> IO ()
-printChecks checks = do
-    mapM_ (putStrLn . formatCheck) $ sort checks
-    if null checks
-        then exitSuccess
-        else die
+toOutputFormat :: String -> Maybe OutputFormat
+toOutputFormat "json" = Just Json
+toOutputFormat "text" = Just Plain
```
[the remark this answers is not in the document]
Comment (lorenzo) [reply app/Main.hs]: I'll use tty, I'm not attached to any names.

**Comment by lorenzo** [comment]
> Is it on purpose that codeclimate is not showing issues detected by shellcheck?

I actually don't know... I need to see this by myself! By no means I'm filtering spellcheck messages

**Comment by lorenzo** [comment]
@zemanlx Ah! I know what is wrong in your test. Codeclimate requires the json errors to be separated by a null byte. `jq` will stop on the first null byte, so you only ever see one error when piping it to `jq`

**Comment by zemanlx** [comment]
@lorenzo Ok, interesting outcome. One of them is doing it wrong :wink:

**Pull request merged**